s3/replication/common/src/s3replicationcommon/jobs.py

```diff
@@ -23,24 +23,40 @@
 
 
 class Jobs:
-    def dumps(obj):
-        """json """
-        return json.dumps(obj._jobs, cls=JobJsonEncoder)
+    def to_json(self):
+        """Converts to json."""
+        return Jobs.to_json(self)
 
     def __init__(self):
         """Initialise jobs collection"""
         # Dictionary holding job_id and fdmi record
         # e.g. : jobs = {"job1": Job({"obj_name": "foo"})}
         self._jobs = {}
 
-    def to_json(self):
-        """Converts to json."""
-        return Jobs.to_json(self)
+    def dumps(obj):
+        """json """
+        return json.dumps(obj._jobs, cls=JobJsonEncoder)
+
+    def get_keys(self):
+        """Returns all jobs"""
+        return self._jobs.keys()
+
+    def get_all_jobs(self):
+        """Returns all jobs with attributes"""
+        return self._jobs.items()
+
+    def clear_jobs(self):
+        """Clear all jobs"""
+        self._jobs.clear()
 
     def count(self):
         """Returns total jobs in collection."""
         return len(self._jobs)
 
+    def update_jobs(self, jobs_dict):
+        """Populdate _jobs dict with another dictionary"""
```

Comment:
Are we updating state of some jobs?

Reply:
Yes, While prefetching, for example 5 jobs, we are going to remove 5 jobs from the 'all jobs' and (dump)update all 5 jobs into 'progressing jobs'.

Reply:
We can call it add_job

Reply:
changed name to add_jobs. done.